File: Practico4/P4Ejercicio1/Python/main_window.py
from PyQt6.QtWidgets import QApplication, QMainWindow, QMessageBox, QInputDialog, QPushButton, QVBoxLayout, QWidget
from PyQt6.QtGui import QScreen

class MainWindow(QMainWindow):
    def __init__(self):
        super().__init__()
        
        self.resize(400, 200)
        self.center()

        self.setWindowTitle("Mi aplicación")

        layout = QVBoxLayout()
        widget = QWidget()
        widget.setLayout(layout)
        self.setCentralWidget(widget)

        boton_confirmar = QPushButton("Confirmar eliminación")
        boton_confirmar.clicked.connect(self.eliminar_usuario)
        layout.addWidget(boton_confirmar)
        
        boton_leer_entrada = QPushButton("Leer entrada")
        boton_leer_entrada.clicked.connect(self.leer_entrada)
        layout.addWidget(boton_leer_entrada)

    def eliminar_usuario(self):
        dialogo = QMessageBox()
        dialogo.setWindowTitle("Confirmación")
        dialogo.setText("¿Está seguro que quiere dar de baja al usuario?")
        dialogo.setStandardButtons(QMessageBox.StandardButton.Ok | QMessageBox.StandardButton.Cancel)
        response = dialogo.exec()

        dialogo = QMessageBox()
        dialogo.setWindowTitle("Resultado")
        dialogo.setText("Usuario dado de baja" if response == QMessageBox.StandardButton.Ok else "Operación cancelada")
        dialogo.exec()
        
    def leer_entrada(self):
        text, ok = QInputDialog.getText(self, "Ingrese Texto", "Por favor ingrese un texto:")

        if ok and text:
            msg_box = QMessageBox()
            msg_box.setWindowTitle("Texto Ingresado")
            msg_box.setText(f"Usted ingresó: {text}")
            msg_box.exec()
            
    def center(self):
        geometry = self.frameGeometry()
        center = QScreen.availableGeometry(QApplication.primaryScreen()).center()
        geometry.moveCenter(center)
        self.move(geometry.topLeft())

File: Practico4/P4Ejercicio1/Java/com/example/main/MainWindow.java
package com.example.main;

import javax.swing.JFrame;
import javax.swing.JPanel;
import javax.swing.BoxLayout;
import javax.swing.JButton;
import javax.swing.JOptionPane;
import java.awt.BorderLayout;
import java.awt.Dimension;
import java.awt.Toolkit;
import java.awt.event.ActionEvent;
import java.awt.event.ActionListener;

public class MainWindow extends JFrame {

    public MainWindow() {
        setTitle("Mi aplicación");
        setSize(400, 200);
        setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
        setLayout(new BorderLayout());

        centerWindow();

        JPanel panel = new JPanel();
        panel.setLayout(new BoxLayout(panel, BoxLayout.Y_AXIS));

        JButton btnConfirmar = new JButton("Confirmar eliminación");
        btnConfirmar.addActionListener(new ActionListener() {
            @Override
            public void actionPerformed(ActionEvent e) {
                eliminarUsuario();
            }
        });
        panel.add(btnConfirmar);

        JButton btnLeerEntrada = new JButton("Leer entrada");
        btnLeerEntrada.addActionListener(new ActionListener() {
            @Override
            public void actionPerformed(ActionEvent e) {
                leerEntrada();
            }
        });
        panel.add(btnLeerEntrada);

        add(panel, BorderLayout.CENTER);
    }

    private void eliminarUsuario() {
        int response = JOptionPane.showConfirmDialog(this,
                "¿Está seguro que quiere dar de baja al usuario?",
                "Confirmación",
                JOptionPane.OK_CANCEL_OPTION,
                JOptionPane.QUESTION_MESSAGE);

        JOptionPane.showMessageDialog(this,
                response == JOptionPane.OK_OPTION ? "Usuario dado de baja" : "Operación cancelada",
                "Resultado",
                JOptionPane.INFORMATION_MESSAGE);
    }

    private void leerEntrada() {
        String input = JOptionPane.showInputDialog(this,
                "Por favor ingrese un texto:",
                "Ingrese Texto",
                JOptionPane.PLAIN_MESSAGE);

        if (input != null && !input.isEmpty()) {
            JOptionPane.showMessageDialog(this,
                    "Usted ingresó: " + input,
                    "Texto Ingresado",
                    JOptionPane.INFORMATION_MESSAGE);
        }
    }

    private void centerWindow() {
        Dimension screenSize = Toolkit.getDefaultToolkit().getScreenSize();
        Dimension frameSize = getSize();
        int x = (screenSize.width - frameSize.width) / 2;
        int y = (screenSize.height - frameSize.height) / 2;
        setLocation(x, y);
    }
}
